Account details api web

MawsFr opened the pull request:
Veuillez corriger mon travail s'il vous plaît messieurs les ministres.

Il me reste à afficher les details dans la page de profil

Github #14

mvaneindie committed:
<chore> Application du thème du projet
Changement du fichier theme/variable.scss et de certains css ne
respectant pas le theme

mvaneindie committed:
<chore> Traduction de certains textes gardés en anglais par ionic
Changement des fichiers i18n afin de garder le système de traduction

mvaneindie committed:
<chore> Ajout d'images pour représenter l'application

mvaneindie committed:
<chore> Changement du widget id du projet

mvaneindie committed:
<feat>Inscription
Mise en place du système d'inscription, sans la connexion

mvaneindie committed:
<feat>Connexion Ajout de la connexion à la page d'inscription et implémentation d'un appel à une api lors de la connexion

Commit:
<feat> Inscription [Test]
-Ajout des tests de création d'utilisateur et get d'un utilisateur
-MAJ POM pour les dépendances de test

Commit:
<chore> Inscription [Securité]
-Début de mise en place des appels sécurisés (autorisation d'appel que sur /api/user en POST et blocage du reste par exemple)
-Encryptage du mot de passe de la création d'un utilisateur
-Maj doc (création d'utilisateur)
-Maj POM (dépendence pour la sécurité)

mvaneindie committed:
<feat>Détails du profils - fonctionnel sans intégration des rangs
Ajout d'une page dédiée aux profils
Ajout d'un composant 'progressbar'
Mise à jour du css

mvaneindie committed:
<refactor>Refonte du code des pages de connexion et d'inscription pour
correspondre à ce qui a été fait pour la page des détails du compte

Commit:
<chore> API [Sécurité]
-Mise en place de la sécurisation des appels vers l'API (Token JWT,filtrage des urls,vérification en base)
-Mise à jour de la data FYTCUser pour prendre en compte la sécurité (field enable and role)
-Mise à jour du controller FYTCUserController pour attendre du JSON en entrée de createUser (au lieu des paramètres d'url)
-Ajout de la dépendence JWT dans le pom API

mvaneindie committed:
<feat>Details d'un compte
Ajout d'une page reprenant toutes les informations dans le profil d'un
utilisateur
La page n'est pas encore dans le profil

Commit:
<chore> Environnement
-Mise en place de l'environnement de prod (prod properties ) et d'intégration (int properties)
-Ajout test Controller FYTC (oubli)

mvaneindie committed:
<feat>Corrections design

Commit:
<chore> Dockerfile
-Création des Dockerfile Intégration et Test

erwandelbrouqueprial committed:
ajout de l'architecture front du site vitrine et implémentation du composant de connexion et celui d'inscription en cours.

erwandelbrouqueprial committed:
modification du composant inscription pour gérer des erreurs personnalisées

erwandelbrouqueprial committed:
ajout de l'affichage de droite dans la vue inscription + ajout du service
authentification, reste à faire l'appel au back

MawsFr committed:
Merge pull request #13 from FEMMLille/connexion_inscription

Connexion inscription

mvaneindie committed:
<feat> Menu principal
Le menu principal est designé et en attente de contenu

erwandelbrouqueprial committed:
redirection sur la page de profile

erwandelbrouqueprial committed:
redirection sur la page de profile

MawsFr committed:
Création du module correspondant à la page de profile

- Ajout d'un dossier account-details dans l'application
- Configuration du routing
- Ajout du module dans l'application

Github #14

MawsFr committed:
Création du module correspondant à la page de profile

- Ajout d'un dossier account-details dans l'application
- Configuration du routing
- Ajout du module dans l'application

Github #14

mvaneindie committed:
<chore> Suppression des fichiers générés

Commit:
<feat> Pull request [Update]
Prise en compte des modifications sur la pull request

Commit:
<feat> Pull request [Role]
-Mise à jour du role utilisé par spring security et par nous (user,admin,agent de la mairie)
-MAJ indentation

erwandelbrouqueprial committed:
Merge branch 'develop' of https://github.com/FEMMLille/findYourTrashCan into connexion_inscription

MawsFr committed:
Merge branch 'develop' into details-compte-mobile

MawsFr committed:
<merge> Merge pull request #15 from FEMMLille/details-compte-mobile

Details compte mobile

erwandelbrouqueprial committed:
<fix> fixation des erreurs de builds
resolution de l'erreur de binding dans le formulaire de connexion

erwandelbrouqueprial committed:
Merge remote-tracking branch 'origin' into connexion_inscription

MawsFr committed:
Merge pull request #16 from FEMMLille/connexion_inscription

Connexion inscription

MawsFr committed:
<merge>Merge branch 'develop' of
https://github.com/FEMMLille/findYourTrashCan into
account-details-api-web

# Conflicts:
#	findyourtrashcan-web/src/main/app/package-lock.json
#	findyourtrashcan-web/src/main/app/proxy.conf.js
#	findyourtrashcan-web/src/main/app/src/app/app.module.ts
#	findyourtrashcan-web/src/main/app/src/app/core/core.module.ts
#	findyourtrashcan-web/src/main/app/src/app/core/shell/header/login/authentication.service.ts
#	findyourtrashcan-web/src/main/app/src/app/core/shell/header/login/login.component.ts
#	findyourtrashcan-web/src/main/app/src/app/core/shell/header/login/login.module.ts
#	findyourtrashcan-web/src/main/app/src/app/core/shell/header/login/login.template.html
#	findyourtrashcan-web/src/main/app/src/app/inscription/inscription.component.ts
#	findyourtrashcan-web/src/main/app/src/translations/en-US.json
#	findyourtrashcan-web/src/main/app/src/translations/fr-FR.json

MawsFr committed:
Merge pull request #17 from FEMMLille/inscription-api

Inscription api

Commit:
Merge branch 'develop' into inscription-api

Commit:
<chore> merge + delete test
Merge + delete d'un test

MawsFr committed:
Merge pull request #18 from FEMMLille/inscription-api

Pull request [Correction test]

MawsFr committed:
Delete FYTCUserControllerTest.java

DarkShad committed:
Merge pull request #19 from FEMMLille/MawsFr-patch-1

Delete FYTCUserControllerTest.java

MawsFr committed:
[USER_U_003] - Détails du compte
Ajout de la page détails de compte + Merge

Github #14

MawsFr committed:
Merge branch 'develop' of https://github.com/FEMMLille/findYourTrashCan into account-details-api-web

Commit:
<refactor> bug properties
Correction bug properties

MawsFr committed:
Merge remote-tracking branch 'origin/account-details-api-web' into
account-details-api-web

Conflicts:
	findyourtrashcan-web/src/main/app/src/app/account-details/account-details.component.ts
	findyourtrashcan-web/src/main/app/src/app/app.module.ts

erwandelbrouqueprial committed:
<fixe> correction du service authentification pour pouvoir se connecter au back

MawsFr committed:
Merge remote-tracking branch 'origin' into account-details-api-web

MawsFr committed:
<fix> [USER_U_003] - Détails du compte

Correction de bugs

Github #14

erwandelbrouqueprial committed:
<feat> Créer le site vitrine modification de la page home + ajout des onglets dans le menu + ajout d'images

MawsFr committed:
[USER_U_003] - Détails du compte + <fix> Ajout d'un profil test

- Création de la page de profil Web 
- Ajout d'un profil test pour que les test s'executent dans une base h2

Github #14

MawsFr committed:
[USER_U_003] - Détails du compte

- Ajout d'un fichier logback.xml
- Ajout de webservice pour la création de Account details et ajout de
ceux dans spring security

Github #14

MawsFr committed:
<chore> Git ignore application-dev.properties

MawsFr committed:
<docs> [USER_U_003] - Détails du compte

- Ajout de la documentation
- Refactoring

Github #14

MawsFr committed:
<chore> Changeemnt du tag pour les corrections bug

MawsFr committed:
Merge remote-tracking branch 'origin/develop' into
account-details-api-web

Conflicts:
	findyourtrashcan-web/src/main/app/src/app/app.module.ts
	findyourtrashcan-web/src/main/app/src/app/home/display/display.module.ts

MawsFr committed:
[USER_U_003] - Détails du compte

- Ajout des champs du profil utilisateur

Github #14

erwandelbrouqueprial commented on findyourtrashcan-web/src/main/app/src/app/account-details/account-details.service.ts:
```diff
@@ -0,0 +1,24 @@
+export class AccountDetailsService {
+    // _user: any;
+
+    // constructor(public api: AuthenticationService) { }
```
dead code ?

erwandelbrouqueprial commented on findyourtrashcan-web/src/main/app/src/app/shared/model/account-details.ts:
```diff
@@ -0,0 +1,13 @@
+// import { RankType } from './rank_type';
+// import { Rank } from './rank';
+// export class AccountDetails {
```
dead code ? sinon faudrait faire une interface de l'objet pour qu'on soit sur qui ai toujours la même forme exemple: export interface InscriptionContext {
    login: string;
    password: string;
    repeatPassword: string;
    email: string;
    firstName: string;
    lastName: string;
    birthday: Date;
    gender: boolean;
}

erwandelbrouqueprial committed:
modification des icones du menu pour correspondre à notre projet

Fcornaire commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/repository/AccountDetailsRepository.java:
```diff
@@ -1,14 +1,22 @@
+	 *            Id of the user we want account details
+	 * @return The account details
+	 */
+	AccountDetails findByUserId(Integer id);
```
Normalement t'a pas besoin de faire ça, la méthode findOne du repository fais la même chose

Fcornaire commented on findyourtrashcan-api/src/main/java/fr/femm/findyourtrashcan/security/WebSecurityConfig.java:
```diff
@@ -34,6 +35,7 @@ protected void configure(HttpSecurity http) throws Exception {
 	    	.csrf().disable()
 	    	.authorizeRequests()
 	    		.antMatchers("/").permitAll()
+				.antMatchers(API_ACCOUNT_DETAILS_URL).permitAll()
```
Donc on a pas besoin d'être connecté pour faire les requêtes sur /api/accountdetails ?

ExSoldat replied:
Je pense qu'on a besoin d'aide sur ça @Darkshad parce que j'ai fait la même pour tester mes services mais on ne devrait pas faire comme ça

MawsFr replied:
bah juste après y'es écrit "authenticated()" donc je pense qu'il faut obligatoirement un token 
et tfacon ca marche pas pour l'instant les appel depuis le client j'ai un 403 uneauthorized c'est que spring security fonctionne
après je sais pas si c'est la bonne méthode mais ca marche

erwandelbrouqueprial committed:
enlevement du service chucknorris

MawsFr committed:
[USER_U_003] - Détails du compte

- Ajout des validateurs sur les champs du formulaire de modifications de
profil
- Ajout du service communiquant avec l'api accountdetails

Github #14

erwandelbrouqueprial committed:
insertion de la communication dans la page home ainsi que sa mise en page, reste 1 carte de communication à mettre pour finaliser la page

erwandelbrouqueprial committed:
reste a centrer le texte sur la page d'accueil, création du component support

MawsFr committed:
[USER_U_003] - Détails du compte

-test

Github #14

MawsFr committed:
<merge> remote-tracking branch 'origin/site_vitrine' into
account-details-api-web

Conflicts:
	findyourtrashcan-web/src/main/app/src/translations/fr-FR.json

MawsFr committed:
<chore> Correction de probleme de build

erwandelbrouqueprial committed:
commencement de l'affichage visuel de la page qui nous sommes

erwandelbrouqueprial committed:
insertion du contenu de la page qui sommes nous, reste à faire la déscription de chacun des collaborateurs

erwandelbrouqueprial committed:
ajout des photos des collaborateurs

erwandelbrouqueprial committed:
modification de l'image du telephone sur la page d'accueil pour avoir un premier visuel de l'appli

erwandelbrouqueprial committed:
suppression du composant support, ajout du composant footer

MawsFr committed:
[USER_U_003] - Détails du compte + <fix> Ajout d'un profil test

- Ajout des objet du modele Role, User, Account details
- Ajout du formulaire et du service
- Ajout de l'api pour faire des requete sur le back

Github #14

MawsFr committed:
<merge> remote-tracking branch 'origin/site_vitrine' into
account-details-api-web

ExSoldat commented on .gitignore:
```diff
@@ -300,4 +301,4 @@ scripts/e2e/webpackEntryPoints.json
 
-findyourtrashcan-api/src/main/resources/application-dev.properties
\ No newline at end of file
+findyourtrashcan-api/src/main/resources/application-dev.propertiesapplication-dev.properties
```
Un peu étrange comme nom non ?

MawsFr replied:
Oui je sais pas pk elle se remet tooujours comme ca :/

MawsFr committed:
[USER_U_003] - Détails du compte + <fix> Ajout d'un profil test

- Ajout du authent.interceptor qui permet d'ajouter le token dans toute
les requetes
- 

Github #14

MawsFr committed:
[USER_U_003] - Détails du compte + <fix> Ajout d'un profil test

- Ajout du authent.interceptor qui permet d'ajouter le token dans toute
les requetes
- 

Github #14

MawsFr committed:
[USER_U_003] - Détails du compte + <fix> Ajout d'un profil test

- Ajout du authent.interceptor qui permet d'ajouter le token dans toute
les requetes
- 

Github #14

MawsFr committed:
[USER_U_003] - Détails du compte + <fix> Ajout d'un profil test

- Ajout de agular2-jwt
- Envoi du token dans toute les requete
- Continuationdu formulaire du detail de compte

Github #14

erwandelbrouqueprial committed:
tentative de correction de l'auth

erwandelbrouqueprial committed:
Merge branch 'account-details-api-web' into site_vitrine

# Conflicts:
#	findyourtrashcan-web/src/main/app/src/app/core/core.module.ts

erwandelbrouqueprial committed:
Merge branch 'account-details-api-web' into site_vitrine

# Conflicts:
#	findyourtrashcan-web/src/main/app/src/app/core/core.module.ts

erwandelbrouqueprial committed:
refont du web

erwandelbrouqueprial committed:
correction pour build

MawsFr committed:
<chore> Fusion de account details avec le nouveau socle

MawsFr committed:
[USER_U_003] - Détails du compte

- Sauvegarde fonctionnelle

Github #14

MawsFr committed:
[USER_U_003] - Détails du compte

- Sauvegarde fonctionnelle

Github #14
